### PR TITLE
Add new Tests

### PR DESCRIPTION
**Pull request by NikkelM**
This PR adds tests for almost all untested methods (not counting UI), as well as refactors some existing tests to make better use of setUp etc.

### CODE REVIEW

File: packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st
Comment (Skn0tt) [inline packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: [sand] Creating the question-to-test in setup makes the tests somewhat hard to read, as you have to jump around the class and do some mental work to figure out what question is actually being tested. AFAICT, local variables inside of the tests would work just fine. What am I missing?

File: packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st
Comment (Skn0tt) [inline packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st:5]: [boulder] I think we should reset in either setUp or Teardown, but not both - double-resetting might hide valid bugs.

File: packages/Liquid-Tests.package/LQPollDraftTests.class/instance/setUp.st
Comment (Skn0tt) [inline packages/Liquid-Tests.package/LQPollDraftTests.class/instance/setUp.st:5]: [sand] Same as above. Are instance variables the way to go?

File: packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st
Comment (Skn0tt) [inline packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st:4]: [sand, not-scope-of-this-pr] we need to do this quite a lot, right? maybe there's some space for abstraction here, where we could extract `PollRepo default:` or `LiquidParticipantMenu answeredPollIds:  Set new` into one / multiple common test classes.

File: packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st
Comment (Skn0tt) [inline packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st:10]: while `isKindOf:` is considered a bad-practice, Squeak is missing an `isUUID` handler on `UUID` :(

Comment (Skn0tt) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st:10]: [dust] (maybe put the above in a comment)

File: packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st
Comment (Skn0tt) [inline packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st:2]:
```suggestion
testIDsAreUUIDsSoTheyDontCollide
```
[dust] would make the comment below obsolete

File: packages/Liquid-Tests.package/LQPollTests.class/instance/testStartTime.st
Comment (Skn0tt) [inline packages/Liquid-Tests.package/LQPollTests.class/instance/testStartTime.st:11]: [dust] this sounds like one could add an `assert:` abstraction to better name this! `self assert: poll startTime isBetween: timeBeforeStart and: timeAfterStart`

or maybe something using intervals: `self assert: poll startTime isIn: (timeBeforeStart to: timeAfterStart)` or `self assert: ((timeBeforeStart to: timeAfterStart) includes: poll startTime)`

(code snippets are rough ideas, they most likely don't work that way)

File: packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st
Comment (Skn0tt) [inline packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st:2]: I like this test! Makes it easy to see if SW was installed properly

**Review by Skn0tt — CHANGES_REQUESTED** [review]
<3 this PR! really good work on adding all of these tests. left a few comments, only one blocking.

File: packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st:5]: As far as I can see @PaulVII wrote this test. After thinking about it for a short while, I think the reset in setUp is done to obviously clear the Set for the test, but I believe they reset it again in tearDown to make sure it doesn't interfere with the "normal" Liquid, since the Poll-ID's used in the test would be stored indefinitely, or until running setUp again. Correct me if I'm wrong

File: packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: Since we need a `LiquidChoicesQuestion` in all tests of this class, I decided to move the initialization of this variable to setUp. We could of course get the same behaviour by using a temporary variable, but I think it is more "idiomatic" (or whatever this falls under) to use setUp in this case - we could however choose a better fitting name for the variable, if you like

File: packages/Liquid-Tests.package/LQPollDraftTests.class/instance/setUp.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollDraftTests.class/instance/setUp.st:5]: My comment above also applies here.

File: packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st:4]: Correct, I also had some problems with `PollRepo default` and finding out when we need to "reset" it. Would you mind to explain your idea further? What do you mean by abstracting the behaviour? We cannot just define a new class `LiquidTestPollRepo` and use `LiquidTestPollRepo default`, since the call to `PollRepo default` is not done in our tests, but within e.g. `LiquidPoll>>startWithId: ifTaken:`.

File: packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st:10]: Exactly my thoughts, good point with the comment, I have included this in the test.

File: packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st:2]: I have changed the naming, good suggestion

File: packages/Liquid-Tests.package/LQPollTests.class/instance/testStartTime.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/testStartTime.st:11]: Good idea. I have checked what we can do, and surprisingly `DateAndTime` is a subclass of `Magnitude`, allowing us to use `between: and:` to basically write exactly what your first suggestion is.

File: packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st:2]: Yep, since we had some problems in the past with SW not being installed for some, I thought it would be nice to have a test for that, especially since this also makes sure SW is installed when the CI is running, meaning the setup in the baseline is correct.

File: packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st:2]: Just to test out how exactly the "suggest changes feature" works - I have marked this comment as resolved, but as you can see below, the suggested change is still blocking the Review/Merge, since I did not apply it from within this comment. However, since the conversation has been marked as resolved, the change cannot be committed anymore (mouseover will say this as a tooltip). Now I see two possibilities:
1: You have the option to change the type of your review from "suggest changes" to "approve" - then my question would be: Why do you need to do this manually?!?
2: We can delete this comment, which is totally counter-intuitive to the "resolve"-feature. If the "skn0tt requested changes" badge below stays after deleting the comment we know that you need to review the PR again, or change the type of the approval.
Dang, this message got long. I think you get that I don't like the feature...

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/testIDsAreUUIDs.st]: Just pinging @Skn0tt in case you miss the above comment since I marked the conversation as resolved

File: packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st
Comment (PaulVII) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: I also think setting up state in setUp is pretty standard behaviour, and I don't think it really matters in a case that is this simple

File: packages/Liquid-Tests.package/LQLoadTest.class/instance/testTwoHundredUsers.st
Comment (PaulVII) [inline packages/Liquid-Tests.package/LQLoadTest.class/instance/testTwoHundredUsers.st:25]: [sand] Maybe use self fail: aMessage instead of fail to make it easier to find out what's wrong (error message will be displayed in the github CI log)

File: packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st
Comment (Skn0tt) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: if there are reasons for `setUp` other than being "idiomatic", it's fine. we shouldn't follow idioms if they don't improve things, though. having the setUp logic made the test harder to read for me. temporary variables would've allowed to read the test without hopping between functions.

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: I understand your perspective. However, I think that _writing_ tests will be a lot easier when using setUp, since we can skip the whole initialization of "stuff", just as we did e.g. in the ObjectRepoTests when it comes to starting the Server. Of course, in this case it is just assigning one variable, but I still believe it is worth it - especially since you only need to check once what `self question` is, but save yourself two lines in every future test

File: packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st
Comment (Skn0tt) [reply packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st:5]: if every test that depends on `LiquidParticipantMenu answeredPollIds` reset it during `setUp`, that should *technically* be enough. if we also reset during `teardown`, that could result in a future test forgetting to clean their dependencies but still passing. resetting twice is defensive programming

File: packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st
Comment (Skn0tt) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st:4]: We could have an abstract `LiquidTestCase` class that resets the PollRepo as part of its setup, which our actual tests could extend from

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUpRemoteServer.st
Comment (PaulVII) [inline packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUpRemoteServer.st:4]: [pebble] What happens if tearDown wasn't executed last time? Will there be problems with just overwriting the existing one? I personally think we should generally also cleanup in setUp to be safe, but I don't know what is the recommended way of doing this, I just remember reading it somewhere once. Julian and me also wondered about this while writing tests, what do you think?

File: packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st:5]: I agree. Especially since we assume that our Users don't touch code, meaning they will not run the tests, we should not have any problems with only resetting in setUp. I however don't see what you mean with 

> that could result in a future test forgetting to clean their dependencies but still passing

How would leaving out tearDown result in such a test failing when forgetting to clean up behind it?

File: packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st:4]: I see. This could indeed solve some problems. Since is a "bigger" change, I would like to get @PaulVII's opinion, I personally would approve

File: packages/Liquid-Tests.package/LQLoadTest.class/instance/testTwoHundredUsers.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQLoadTest.class/instance/testTwoHundredUsers.st:25]: +1 from me too, I will add some messages to all occurrences of fail

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st
Comment (PaulVII) [inline packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st:9]: [sand] Theres also self should: aBlock raise: anException , which would be cleaner here IMO. 
[boulder] This test passes if no error is thrown, as I understand it this is not the intended behaviour. Could be fixed by using the suggestion above

File: packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st
Comment (PaulVII) [reply packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st:5]: Yes, we did that, I also added a comment on some other file explaining why and asking for feedback^^

File: packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st
Comment (PaulVII) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st:4]: Maybe we could have a LQTestUtils class or similar with setupLocalRepo and similar methods? But I'd also be fine with keeping it as it is

File: packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st
Comment (Skn0tt) [reply packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st:5]: Assume a test cases A that requires some shared global state to be blank for its tests to pass. It implements `setUp` and `teardown` to reset global state.

Now another developer writes a new test case B, it too depends on the same global state - but they're not aware of it. Since `A >> tearDown` cleared the state for them, everything's fine - as long as A runs before B. Once the execution order changes, or B is run two times in a row, it fails.

If A didn't implement `tearDown`, then the developer of `B` would've been more likely to stumbled over the shared global state, and would've remembered to implement `setUp`  themselves.

Thus, counterintuitively, implementing `tearDown` helps hiding faulty test code instead of making it obvious.

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st:5]: Top, thanks for the example. I was still mentally "locked up" in the current class, where we clear the global state in setUp anyways. Still, maybe @PaulVII wants to add his opinion before we change this.

File: packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st
Comment (PaulVII) [inline packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st:1]: [sand] unclassified. Maybe we should run the linter on Liquid-Tests as well to prevent stuff like this? That would require some exceptions (e.g. LiquidLintTests has too many methods)

**Review by PaulVII — CHANGES_REQUESTED** [review]
Looking very good! I don't approve for now bc i think one test is not working as desired, but I'm overall very impressed and think this PR massively improves our project!

File: packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st
Comment (PaulVII) [reply packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st:5]: As I understand it, @Skn0tt advocates for just using setUp and never teardown right? I'd be fine with that, it solves my problem too^^

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQParticipantMenuTests.class/instance/tearDown.st:5]: Done, I have deleted the tearDown-Method

File: packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st:4]: What do you mean by "keeping it as it is"? Setting PollRepo default in setUp of each test that neds it? And if we add a superclass `LiquidTestCase` I would just implement setUp, there which just sets PollRepo default to a new LocalObjectRepo. Tests which need a RemoteServer will need to do that themselves anyways

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st:1]: As I understand it you would define a superclass `LQObjectRepoTests` with two subclasses `LQLocalObjectRepoTests` and `LQRemoteObjectRepoTests`, which implement all of the tests for their respective ObjectRepo? I would second that opinion, this makes it easier to define which test uses which ObjectRepo, plus we can make sure to test all tests on both Repos.

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUpRemoteServer.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUpRemoteServer.st:4]: I don't know about doing cleanup in setUp.
LQPortAssigner is not affected by setUp or tearDown, but just always counts up the Ports, with each `get` it returns the next port in its list. Is this what you meant?

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st]: [sand]: You would replace line 9 with your suggestion, do I understand correctly? That way, you would circumvent `on: do:` by using `should: raise:` which would indeed be cleaner.
[boulder]: The test should not pass if no error is thrown, did you test this? dummy escalate throws a RemoteError, which leads us going into the `do:`-Block, which is what we wanted. If we pass the `do:` we `self fail` in the next line.

File: packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st:1]: Hmm, I'm not sure. There are pros and cons to that decision for sure. You know best about the Linter, how much work would that change be?

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUpRemoteServer.st
Comment (PaulVII) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUpRemoteServer.st:4]: Ahh sorry, didn't see the self fail in the last line, so this is purely a cosmetic suggestion now: replace 
[dummy escalate] on: RemoteError do: [^ self "everything is good" ].
	self fail
with
	self
		should: [dummy escalate]
		raise: RemoteError
I tested it, it runs through

**Review by PaulVII — APPROVED** [review]

File: packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st
Comment (PaulVII) [reply packages/Liquid-Tests.package/LQStatisticsWorkbenchIntegrationTests.class/instance/testStatisticsWorkbenchIsDefined.st]: I'll try that somewhere else. Consider this discussion solved :)

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st]: Adding to the boulder: I think I see what you mean now. `dummy escalate` indeed does not throw a `RemoteError`, but a general Error - that is a mistake on my part. I have changed this and also `should: raise:`

Comment (PaulVII) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st]: Ahh sorry, didn't see the self fail in the last line, so this is purely a cosmetic suggestion now: replace 
```smalltalk
[dummy escalate] on: RemoteError do: [^ self "everything is good" ].
	self fail
```
with
```smalltalk
	self
		should: [dummy escalate]
		raise: RemoteError
```
I tested it, it runs through.
Since the boulder is solved, I approved the PR

Comment (PaulVII) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st]: > Adding to the boulder: I think I see what you mean now. `dummy escalate` indeed does not throw a `RemoteError`, but a general Error - that is a mistake on my part. I have changed this and also `should: raise:`

That's not what I meant, but good think you found another error this way ^^

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st]: It seems we commented at the same time ^^

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st
Comment (PaulVII) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st:1]: > As I understand it you would define a superclass `LQObjectRepoTests` with two subclasses `LQLocalObjectRepoTests` and `LQRemoteObjectRepoTests`, which implement all of the tests for their respective ObjectRepo? I would second that opinion, this makes it easier to define which test uses which ObjectRepo, plus we can make sure to test all tests on both Repos.

Yes, thats about what I meant. But is there should be no implementation difference for the tests, right? The only difference is what client they use, so the only change in the subclasses would be defining a different client in setUp

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st]: > > Adding to the boulder: I think I see what you mean now. `dummy escalate` indeed does not throw a `RemoteError`, but a general Error - that is a mistake on my part. I have changed this and also `should: raise:`
> 
> That's not what I meant, but good think you found another error this way ^^

What happened there is that we had `[dummy escalate]` in []-brackets, which evaluated the statement in the block. It seems that that evaluation gets interrupted by the error and just returns (I tested this in a Workspace, `[dummy escalate]` doesn't throw an error...), without passing on the error. I don't however know why the test didn't fail afterwards, when we got to `self fail`

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st:1]: Yes, it would be just like the discussed change of adding a superclass for all tests. Actually, should we implement that idea as well, we wouldn't even need an extra setUp for the LocalObjectRepo-Tests, since that would be done in the superclass.
How I would do it now is to implement all tests in the `ObjectRepoTests`-superclass, and to call `super testNameHere` in the subclasses with their client as argument

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st:1]: @Skn0tt The messagePassing-Test only works on a RemoteRepo, correct? I don't see how we would do it with a LocalRepo

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st]: The newest commit includes this change, @PaulVII please check if this is what you meant.

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/testErrorHandling.st]: > What happened there is that we had [dummy escalate] in []-brackets, which evaluated the statement in the block. It seems that that evaluation gets interrupted by the error and just returns (I tested this in a Workspace, [dummy escalate] doesn't throw an error...), without passing on the error. I don't however know why the test didn't fail afterwards, when we got to self fail

Correction: The code worked as it should have. The difference is that when the client is a `RemoteRepoClient` `dummy escalate` raises a RemoteError, whilst a `LocalObjectRepo` raises a normal Error. My tests seem to have used different Repos, which is why the test didn't work as it would normally. So the "error" that I found wasn't an error, but just an oversight when extending the errorHandling-Test to the LocalObjectRepo

File: packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st
Comment (PaulVII) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st]: > The newest commit includes this change, @PaulVII please check if this is what you meant.

Yes that's about it ^^ I realize now that what I actually wanted is not really possible, since there cannot be an abstract test case class - it's tests would fail. Locally, I made the remote tests inherit from the local ones, this way we could get rid of five or so more duplicate methods. If you think this would be worth it, I can push that as a commit.

grafik
grafik

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st]: Before you do that: Note that currently, your `LQRemoteObjectRepoTests` only has one test. It does not inherit the tests, meaning it does not test the tests implemented in its superclass, if you do not specifically tell it to do so

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st]: It seems that I made a mistake - the tests do actually get inherited, I tested something similar, but different as I just found out. So your proposal could be be a good idea - how many methods would we get rid of?

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st]: Okay, I seem to have found out how we can _actually_ do this: 
1. If a class has tests, it does _not_ inherit the tests from its superclass.
2. If the class does not have any tests, it _does_ inherit the tests from its superclass.

Since No. 2 applies in our case, I had a look into `TestCase` and found that if the superclass is abstract, or the current class does not have any tests, it inherits the selectors of its superclass. By just overwriting the class-side method `shouldInheritSelectors`, we can force `LQRemoteObjectRepoTests` to also perform all the tests of its superclass.
When I tested this (see my last comment), I must have had a class without any tests itself, which is why it performed the tests of the superclass.
The next commit includes this changed class hierarchy, which saves us two methods for each added test performed on both Repo-Types.

File: packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: I added an abstract superclass in the newest commit, please both @PaulVII and @Skn0tt take a look if you like it, else we can remove it again to work on it further

File: packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st
Comment (NikkelM) [reply packages/Liquid-Tests.package/LQPollTests.class/instance/setUp.st]: > I added an abstract superclass in the newest commit, please both @PaulVII and @Skn0tt take a look if you like it, else we can remove it again to work on it further

I put the comment in the wrong conversation :|

File: packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st
Comment (Skn0tt) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: what is "the latest commit"? 😅

Comment (Skn0tt) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: btw: you *can* delete comments and recreate them in the right conversation

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: The one saying "Added an abstract superclass" ^^

Comment (NikkelM) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: > btw: you _can_ delete comments and recreate them in the right conversation

Of course, but I didn't want to just delete it here, in case someone already saw it, as I noticed quite late, so I just added the note and copied the comment over to the correct conversation

**Review by Skn0tt — APPROVED** [review]
Unsure which of the conversations require my answer. I'm fine with resolving them @NikkelM :)

File: packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st
Comment (Skn0tt) [reply packages/Liquid-Tests.package/LQChoicesQuestionTests.class/instance/setUp.st:4]: d3c61e6 you mean?

**Pull request merged**
